src/components/TileCatalogNew/TileCatalogNew.jsx

```diff
@@ -60,60 +70,57 @@ const TileCatalogNew = ({
   onSort,
   sortOptions,
   selectedSortOption,
+  isLoading,
+  error,
   i18n,
+  className,
 }) => {
   const [currentPage, setCurrentPage] = useState(1);
-  const [tileHeight, setTileHeight] = useState(0);
 
-  const onSize = ({ height }) => {
-    setTileHeight(height);
-  };
-
-  const HeightMonitor = sizeMe({
-    monitorHeight: true,
-  })(({ content }) => <div>{content}</div>);
-
-  const getTile = (rowIdx, colIdx) => {
-    const numTilesPerPage = numRows * numColumns;
-    const tileIndex = rowIdx * numColumns + colIdx + (currentPage - 1) * numTilesPerPage;
-
-    if (tileIndex === 0) {
-      return <HeightMonitor content={tiles[0]} onSize={onSize} />;
-    }
-    return tiles[tileIndex] !== undefined ? (
-      tiles[tileIndex]
-    ) : (
-      <div style={{ height: tileHeight }} />
-    );
-  };
+  // determine how many tiles to render per page
+  const isInCurrentPageRange = i =>
+    i < numColumns * numRows * currentPage && i >= numColumns * numRows * (currentPage - 1);
 
   const renderGrid = () => (
-    <div className="bx--grid">
-      {Array(numRows)
-        .fill(null)
-        .map((i, rowIdx) => (
-          <div className="bx--row">
-            {Array(numColumns)
-              .fill(null)
-              .map((j, colIdx) => (
-                <div className="bx--col">{getTile(rowIdx, colIdx)}</div>
-              ))}
-          </div>
-        ))}
+    <div
+      className={`${iotPrefix}--tile-catalog--grid-container`}
+      style={{
+        '--columns': minTileWidth
+          ? // if the user specifies a minTileWidth, we will render all tiles on one responsive page
+            `repeat(auto-fill, minmax(${minTileWidth}, 1fr))`
+          : // if the user specifies numColumns we will render exactly that number
+            `repeat(${numColumns}, 1fr)`,
+      }}
```

Comment:
I think we can put this in a .scss file by using the `minTileWidth` to set a custom property to be used in `repeat()`.

An additional issue though - in Firefox `repeat(auto-fill, ...)` and `repeat(auto-fit, ...)` only support one repeated column (see [bug 1341507](https://bugzil.la/1341507)).

Reply:
Looks like that bug was resolved / fixed 2 months ago @tay1orjones 

Side note: I'm surprised they created their own issue tracker

Reply:
I think the Firefox support should be fine here, because this catalog (in the case of the user specifying a minTileWidth) is only repeatedly outputting one column anyway. It takes the tiles and just spits them out onto a single paged, responsive grid, treating them all the same by displaying the components as they are recieved.
My understanding is that the limitation here is that we can't specify two column sizes in Firefox like this: `repeat(auto-fill, minmax(${minTileWidth}, 50px, 75px))`

Reply:
In regards to the `--columns` variable, I'm not sure I follow. It looks to me like I have satisfied this suggestion. I may be misunderstanding?

Reply:
Well hey, yeah they did fix it recently! MDN is outdated then 😮I just copied and pasted the support note from there.

@JoelArmendariz Sorry - Yes, the custom property is fine. I think the `repeat()`'s should be in a class in the .scss file if possible, rather than inlined in the `style`

Reply:
Recapping our offline convo - I think it's fine as is. `repeat()` syntax is something I've not seen much of and tripped me up 😅 

I think it's valuable to keep the context of the `repeat()` function in the component rather than the `.scss`. Either way it's going to be used in the `grid-template-columns` in the sass.